Sources/KDTree.swift

```diff
@@ -56,21 +56,26 @@ extension KDTree {
         // copy values from the array
         pointer.initialize(from: values, count: count)
         
-        self = KDTree(values: pointer, startIndex: 0, endIndex: count, depth: depth)
+        self = KDTree(values: pointer, startIndex: 0, endIndex: count, depth: depth, dimensionsOverride: dimensionsOverride)
         
         // deallocate the pointer
         pointer.deallocate()
     }
     
-    private init(values: UnsafeMutablePointer<Element>, startIndex: Int, endIndex: Int, depth: Int = 0) {
+    private init(values: UnsafeMutablePointer<Element>, startIndex: Int, endIndex: Int, depth: Int = 0, dimensionsOverride: Int? = nil) {
         guard endIndex > startIndex else {
             self = .leaf
             return
         }
         
         let count = endIndex - startIndex
         
-        let currentSplittingDimension = depth % Element.dimensions
+        var usedDimensions = Element.dimensions
```

Comment:
I think the more idiosyncratic way of writing this would be:

```
let usedDimensions = dimensionsOverride ?? Element.dimensions
```

I'd also remove the `print` in the resulting production code :)

Reply:
err yes the print has to go :). 
```
let usedDimensions = dimensionsOverride ?? Element.dimensions
``` 
I was not aware of this way of writing, honestly I've always used the other notation and it worked. I do indeed prefer your suggestion.

Reply:
yeah, it's similar to the [elvis operator](https://en.wikipedia.org/wiki/Elvis_operator) `?:`, just for optionals in Swift :)